### PR TITLE
Update shared Http3 code

### PR DESCRIPTION
**Pull request by jkotalik**
Shared code needed to be updated from my original HTTP/3 PR.

Fixes 3 different bugs in shared code.

### CODE REVIEW

File: src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using System.Diagnostics;
 using System.Net.Http.HPack;
 using System.Numerics;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
```
Comment (jkotalik) [inline src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs]: This isn't good!

File: src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs [hunk 2]
```diff
@@ -222,7 +223,7 @@ private void OnByte(byte b, IHttpHeadersHandler handler)
                 case State.CompressedHeaders:
                     switch (BitOperations.LeadingZeroCount(b))
                     {
-                        case 0: // Indexed Header Field
+                        case 24: // Indexed Header Field
```
Comment (jkotalik) [inline src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs]: @scalablecory I think both of us fixed this in our PRs. Taking either fix is okay I think.

Comment (jkotalik) [reply src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs]: I changed it to your fix.

File: src/libraries/Common/src/System/Net/Http/Http2/ReadMe.SharedCode.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ aspnet/AspNetCore code paths:
 
 ## Building dotnet/runtime code:
 - https://github.com/dotnet/runtime/tree/master/docs/workflow
-- Run *build.cmd* from the root once: `PS D:\github\runtime> .\build.cmd -subsetCategory libraries`
+- Run libraries.cmd from the root once: `PS D:\github\runtime> .\libraries.cmd`
```
Comment (Tratcher) [inline src/libraries/Common/src/System/Net/Http/Http2/ReadMe.SharedCode.md]: Revert. Runtime's version is correct.

File: src/libraries/Common/src/System/Net/Http/Http3/ReadMe.SharedCode.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ To copy code from aspnet/AspNetCore to dotnet/runtime, set RUNTIME_REPO to the r
 ## Building dotnet/runtime code:
 - https://github.com/dotnet/runtime/blob/master/docs/libraries/building/windows-instructions.md
 - https://github.com/dotnet/runtime/blob/master/docs/libraries/project-docs/developer-guide.md
-- Run *build.cmd* from the root once: `PS D:\github\runtime> .\build.cmd -subsetCategory libraries`
+- Run libraries.cmd from the root once: `PS D:\github\runtime> .\libraries.cmd`
```
Comment (Tratcher) [inline src/libraries/Common/src/System/Net/Http/Http3/ReadMe.SharedCode.md]: Revert. Runtime's version is correct.

File: src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs [hunk 5]
```diff
@@ -500,7 +502,7 @@ private void OnIndexedHeaderField(int index, IHttpHeadersHandler handler)
 
         private static void ThrowDynamicTableNotSupported()
         {
-            throw new QPackDecodingException(SR.net_http_qpack_no_dynamic_table);
+            throw new QPackDecodingException("No dynamic table support");
```
Comment (Tratcher) [inline src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs]: Revert, I'll fix this in aspnet.

**Comment by jkotalik** [comment]
@Tratcher updated.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**